### DISCUSSION TITLE
How to set isolation_level per connection for SQLite?

### DISCUSSION THREAD

**Discussion by embecka** [sqlalchemy/sqlalchemy#7071] [Usage Questions] [answered]
Hi, I am using sqlalchemy 1.4 to connect SQLite database with pysqlite driver.
I found in the [documentation](https://docs.sqlalchemy.org/en/14/dialects/sqlite.html#serializable-isolation-savepoints-transactional-ddl) how to properly set `isolation_level` for engine. It is advised to not use the `Connection.execution_options.isolation_level `setting on `Connection` and `create_engine()` with the SQLite driver, as this function necessarily will also alter the `isolation_level` setting.
Now I am writing an API and I want to use read only `isolation_level` for `GET` method and read-write `isolation_level` for the rest of methods using the same connection pool. How can I do that?

**Comment by CaselIT** [comment]
Hi,

If I understand the request correctly you can create an engine "copy" from an existing one that will share the same pool like this:
```py
orig_engine = create_engine('sqlite://....')
engine_with_options = engine.execution_options(...)
```

**Reply by embecka** [reply]
If I understand it correctly, this is not advised. See the warning note in the [documentation](https://docs.sqlalchemy.org/en/14/dialects/sqlite.html#serializable-isolation-savepoints-transactional-ddl).

**Comment by zzzeek** [comment, marked as answer]
SQLite's "read only" mode is not a product of "isolation level" it's something you'd put in the URL, in the example at https://docs.sqlalchemy.org/en/14/dialects/sqlite.html#uri-connections .

if OTOH you are looking to set the ["read_uncommitted" pragma](https://www.sqlite.org/pragma.html#pragma_read_uncommitted), which is the only other "isolation level" SQLte supports, you would emit this pragma command yourself on the connection:   "connection.execute(text("pragma read_uncommitted=1"))".
